app/models/candidate_interface/gcse_qualification_details_form.rb

```diff
@@ -17,16 +18,39 @@ def self.build_from_qualification(qualification)
     def save_base
       return false unless valid?
 
-      qualification = ApplicationQualification.find(qualification_id)
-
       qualification.update(grade: grade, award_year: award_year)
     end
 
   private
-
     def validate_award_year_is_date
       valid_award_year = award_year.match(/^[1-9]\d{3}$/)
       errors.add(:award_year, :invalid) unless valid_award_year
     end
+
+    def validate_grade_format
+      return if new_record? || qualification.qualification_type.nil?
+
+      qualification_rexp = invalid_grades[qualification.qualification_type.to_sym]
+
+      if qualification_rexp && grade.match(qualification_rexp)
+        errors.add(:grade, :invalid)
+      end
+    end
+
+    def qualification
+      @qualification ||= ApplicationQualification.find(qualification_id)
+    end
+
+    def invalid_grades
+      {
+        gcse: /[^1-9A-GU\*]/i,
+        gce_o_level: /[^A-EU]/i,
+        scottish_higher: /[^A-D1-7]/i,
```

Comment:
in light of https://trello.com/c/Pb9QRUPv/345-change-scottish-higher-to-scottish-national, this line will need an update

Reply:
Here is the PR for that: https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/516